sklearn/model_selection/tests/test_search.py

```diff
@@ -332,6 +333,60 @@ def test_grid_search_groups():
         gs.fit(X, y)
 
 
+def test_grid_search_future_warnings():
+    # Test that warnings are raised. Will be removed in 0.22
+
+    X = np.arange(100).reshape(10, 10)
+    y = np.array([0] * 5 + [1] * 5)
+
+    clf = LinearSVC(random_state=0)
+    grid = {'C': [.1]}
+
+    def init(clf, parameters, return_train_score):
+        grid_search = GridSearchCV(clf, parameters,
+                                   return_train_score=return_train_score)
+        grid_search.fit(X, y)
+        return grid_search
+
+    msg = ("Computing training scores may affect performance "
+    "significantly. This is the reason return_train_score will "
+    "change its default value from True (current behaviour) to "
+    "False in 0.22. Please set explicitly return_train_score to "
+    "get rid of this warning.")
+    assert_warns_message(FutureWarning, msg, init, clf, grid, "warn")
+    assert_true("mean_train_score"
+                in init(clf, grid, "warn").cv_results_.keys())
+    assert_no_warnings(FutureWarning, msg, init, clf, grid, True)
+    assert_no_warnings(FutureWarning, msg, init, clf, grid, False)
+
+
+def test_random_search_future_warnings():
```

Comment:
If I were you i'd just loop over GridSearchCV and RandomizedSearchCV in the previous test.